WcaOnRails/app/views/static_pages/teams_committees.html.erb

```diff
@@ -31,7 +37,19 @@
     <% team.current_members.includes(:user).order(team_leader: :desc).order(team_senior_member: :desc).order("users.name asc").each do |tm| %>
       <div class="badge team-member-badge <%= tm.team_senior_member ? "team-senior-member-badge" : "" %> <%= tm.team_leader ? "team-leader-badge" : "" %>">
         <%= render "shared/user_avatar", user: tm.user %>
-        <div class="team-member-name"><%= tm.user.name %>
+        <% if tm.user.wca_id %>
+          <% if tm.team_senior_member or tm.team_leader %>
+            <%= link_to person_path(tm.user.wca_id), title: t("about.structure.users.profile", user_name: tm.user.name) do %>
+              <div class="team-member-name"><%= tm.user.name %>
+            <% end %>
+          <% else %>
+            <%= link_to person_path(tm.user.wca_id), title: t("about.structure.users.profile", user_name: tm.user.name) do %>
```

Comment:
What's the difference with the other branch of the condition here?
It looks to me both links are identical but I may be missing something obvious.

The pattern "if wca_id then link_with_name else name" seems very repetitive in that PR, I think it would be best to DRY it up and put it in a helper.
Eg adding that function to `app/helpers/static_pages_helper.rb`:
```ruby
def format_team_member_name(user)
  if user.wca_id
      # the link_to code
  else
      # just the name div
  end
end
```

And use `format_team_member(tm.user)` where you would want to display link or name depending on the WCA ID.

Reply:
Due to the div not holding just the name I've done the helper this way:

```
  def format_team_member_content(user, positions_content)
    if user.wca_id # We have a WCA ID we can link to
      content_tag(:div, 
                  link_to(user.name, 
                          person_path(user.wca_id), 
                          title: t("about.structure.users.profile", user_name: user.name), 
                          data: { toggle: "tooltip", placement: "bottom" }) + tag.br + content_tag(:span, positions_content, class: "team-subtext"), 
                  class: "team-member-name")
    else
      content_tag(:div, user.name + tag.br + content_tag(:span, positions_content, class: "team-subtext"), class: "team-member-name") 
    end
  end
```
It might be a little more obscure but the view is way cleaner now:

```
  <div class="officer-container">
    <% @officers.each do |user, positions| %>
      <div class="badge team-member-badge officer-badge">
        <%= render "shared/user_avatar", user: user %>
        <%= format_team_member_content(user, positions) %> 
      </div>
    <% end %>
  </div>

  <% ([Team.board] + Team.all_official + Team.all_councils).each do |team| %>
    <h3><%= team.name %> <%= team != Team.board ? "(" + team.acronym + ")" : "" %> <span class="badge team-email"><%= mail_to team.email %></span></h3>
    <p><%= t("about.structure.#{team.friendly_id}.description") %></p>
    <br />

    <% team.current_members.includes(:user).order(team_leader: :desc).order(team_senior_member: :desc).order("users.name asc").each do |tm| %>
      <% if tm.team_senior_member %>
        <% positions = t("about.structure.senior_member") %>
      <% elsif tm.team_leader %>
        <% positions = t("about.structure.leader") %>
      <% else %>
        <% positions = "" %>
      <% end %>
      <div class="badge team-member-badge <%= tm.team_senior_member ? "team-senior-member-badge" : "" %> <%= tm.team_leader ? "team-leader-badge" : "" %>">
        <%= render "shared/user_avatar", user: tm.user %>
        <%= format_team_member_content(tm.user, positions) %> 
      </div>
    <% end %>
  <% end %>
```

I still don't like the positions if/else to add the "Leader" o "Senior Member" subtext.

Should I go ahead and commit these changes or this is not what you had in mind?

Reply:
I definitely had something in mind, I can suggest something slightly better:
```ruby
  # in helpers
  def badge_for_member(tm)
    if tm.team_leader
      "team-leader-badge"
    elsif tm.team_senior_member
      "team-senior-member-badge"
    end
  end
  def team_member_name(name)
    content_tag(:div, class: "team-member-name") do
      name + tag.br +
      content_tag(:span, class: "team-subtext") do
        yield
      end
    end
  end
  def format_team_member_content(user)
    name = if user.wca_id
             link_to(user.name, person_path(user.wca_id), title: t("about.structure.users.profile", user_name: user.name), data: { toggle: "tooltip", placement: "bottom" })
           else
             user.name
           end
    team_member_name(name) do
      yield
    end 
  end
```
```erb
  # and in the view:
  <% team.current_members.includes(:user).order(team_leader: :desc).order(team_senior_member: :desc).order("users.name asc").each do |tm| %>
      <div class="badge team-member-badge <%= badge_for_member(tm) %>">
        <%= render "shared/user_avatar", user: tm.user %>
        <%= format_team_member_content(tm.user) do %>
          <% if tm.team_senior_member %>
            <%= t("about.structure.senior_member") %>
          <% elsif tm.team_leader %>
            <%= t("about.structure.leader") %>
          <% end %>
        <% end %> 
      </div>
  <% end %>
```

The indent may be off but that's the idea!
The point here is that I pass a block computing the member position to the helper so I can just yield it; I compute how the name should look like to not bloat the content_tag; I reused the code computing the `div.team-member-name` to avoid the duplication.
I also added a helper that just compute the appropriate css class to apply instead of putting lengthy `if`s directly in the tag.

Reply:
I'm happy with what I just committed, I've tried to put the officers in the main loop but the badges are constructed differently and the code looked worse IMHO.

After all the work in this page I have a nice idea of how to improve the Delegates page also, maybe that would be a good idea before cleaning the css stuff.